### PR TITLE
tests: add unit test for `PayloadTooLarge` error

### PR DESCRIPTION
**Pull request by cymruu**
# Pull Request
Add a unit test for the `Payload`, which verifies that a request with a payload that is too large is rejected with the appropriate message.
This was requested in this PR https://github.com/meilisearch/meilisearch/pull/3739

## Related issue
https://github.com/meilisearch/meilisearch/pull/3739

## What does this PR do?
- Adds requested test

## PR checklist
Please check if your PR fulfills the following requirements:
- [ ] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [ ] Have you read the contributing guidelines?
- [ ] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: meilisearch/src/extractors/payload.rs [hunk 1]
```diff
@@ -71,3 +71,40 @@ impl Stream for Payload {
...
+            limit: 3,
+            remaining: 3,
+            payload: Decompress::new(actix_payload, actix_http::ContentEncoding::Identity),
+        };
+
+        let mut enumerated_payload_stream = payload.enumerate();
+
+        while let Some((idx, chunk)) = enumerated_payload_stream.next().await {
+            if idx == 3 {
+                snapshot!(chunk.unwrap_err(), @"The provided payload reached the size limit. The maximum accepted payload size is 3 B.");
```
Comment (cymruu) [inline meilisearch/src/extractors/payload.rs:106]: This probably is not ideal, but currently `MeilisearchHttpError` does not implement `PartialEq`. 
Maybe it should?

File: meilisearch/src/extractors/payload.rs [hunk 2]
```diff
@@ -71,3 +71,40 @@ impl Stream for Payload {
...
+mod tests {
+    use actix_http::encoding::Decoder as Decompress;
+    use actix_http::BoxedPayloadStream;
+    use bytes::Bytes;
+    use futures_util::StreamExt;
+    use meili_snap::snapshot;
+
+    use super::*;
+
+    #[actix_rt::test]
```
Comment (cymruu) [inline meilisearch/src/extractors/payload.rs:85]: I didn't see any other unit tests defined this way. I hope it is not a problem.

File: meilisearch/src/extractors/payload.rs (hunk 1 above)
Comment (Kerollmops) [reply meilisearch/src/extractors/payload.rs:106]: If it simplified the code you wrote there, I suggest you derive `PartialEq` (and maybe even `Eq`) on `MeilisearchHttpError` 😊

File: meilisearch/src/extractors/payload.rs (hunk 2 above)
Comment (Kerollmops) [reply meilisearch/src/extractors/payload.rs:85]: It doesn't look like an issue ☁️

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Thank you very much for your work and time 💟

**Comment by curquiza** [comment]
Hello @cymruu do you need help on this PR to apply what kero asked here?
https://github.com/meilisearch/meilisearch/pull/3745#discussion_r1193893738

**Comment by cymruu** [comment]
@curquiza Hello! Yes, I got stuck on implementing `PartialEq` for `MelisearchHttpError`. The problem is that the error contains variants which include 3rd party errors that do not derive `PartialEq`. (for example `SerdeError`, or `heed:Error`) How should I approach that? The only solution I can think of is to wrap those errors with another struct and implement `PartialEq` for them, but I must be missing something.

My second concern is how to define partially equal logic for error types. For instance when 2 `SerdeErrors` are considered partially equal? Should we compare just its `ErrorCode` or all properties of the error struct? Defining this behavior seems to exceed the scope of this PR and probably requires more consideration. 

Let me know what you think!

**Comment by Kerollmops** [comment]
Hey @cymruu,

I am sorry I thought it would be easy to add but I was wrong. If you do not implement the `PartialEq` trait, is there anything missing?

**Comment by curquiza** [comment]
bors try

**Comment by curquiza** [comment]
@Kerollmops we can merge it when you want

**Comment by Kerollmops** [comment]
Thank you very much for that! And sorry for the delay 🧀

bors merge

**Comment by Kerollmops** [comment]
Bors merge

**Review by Kerollmops — APPROVED** [review]
bors merge

**Comment by curquiza** [comment]
bors try

**Comment by curquiza** [comment]
bors merge-

**Comment by curquiza** [comment]
bors merge

**Pull request merged**